crates/bevy_hierarchy/src/propagatable.rs

```diff
@@ -1,43 +1,77 @@
-use crate::components::{GlobalTransform, Transform};
-use bevy_ecs::prelude::{Changed, Entity, Query, With, Without};
-use bevy_hierarchy::{Children, Parent};
+use bevy_ecs::prelude::*;
 
-/// Update [`GlobalTransform`] component of entities based on entity hierarchy and
-/// [`Transform`] component.
-pub fn transform_propagate_system(
+use crate::{Children, Parent};
+
+/// Marks a component as propagatable thrown hierachy alike `Transform`/`GlobalTransorm`
+/// or `Visibility`/`ComputedVisibility`.
+pub trait Propagatable: Component {
```

Comment:
```suggestion
pub trait Propagate: Component {
```

Naming traits after verbs is usually more idiomatic (and shorter).